File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <vector>
#include <set>
using namespace std;

vector<int> solution(vector<int> numbers) {
    set<int> sums;
    
    int n = numbers.size();
    for(int i = 0; i < n; i++){
        for(int j = 0; j < n; j++){
            if(i == j) continue;
            sums.insert(numbers[i] + numbers[j]);
        }
    }
    
    vector<int> answer(sums.begin(), sums.end());
    return answer;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = {};
        
        int n = numbers.length;
        TreeSet<Integer> set = new TreeSet<>();
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                if(i == j) continue;
                
                set.add(numbers[i] + numbers[j]);
            }
        }
        
        answer = set.stream().mapToInt(Integer::intValue).toArray();
        return answer;
    }
}